crates/types/src/spline_motion_interpolator.rs

```diff
@@ -0,0 +1,107 @@
+use splines::{Interpolation, Key, Spline};
+use thiserror::Error;
+
+use crate::{Joints, MotionFile};
+use std::time::Duration;
+
+pub struct SplineMotionFileInterpolator {
+    interpolator: Spline<f32, Joints>,
+    current_time: Duration,
+    start_time: Duration,
+    end_time: Duration,
+}
+
+#[derive(Error, Debug)]
+pub enum InterpolatorError {
+    #[error("accessed interpolator defined for [{min}s, {max}s] at {current}s")]
+    OutOfBoundsAccess { min: f32, max: f32, current: f32 },
```

Comment:
- Please don't abbreviate 
- Are you using `f32` for time? This smells like [Primitive Obsession](https://refactoring.guru/smells/primitive-obsession).

Reply:
The library has to normalize timesteps. For a time s in [a,b], the normalized time is calculated by (s-a)/(b-a). Durations do not (yet) have a stable division implementation. Therefore we decided to convert to f32 internally, but do the conversion as late as possible. If you have other ideas how to tackle this we would be open for them.

Reply:
Okay, understood. But if I saw it correctly, you are converting it just before instantiating the error. Couldn't you just keep the durations and pass them in the error? There should be no spline stuff involved anymore? 🤔

Reply:
Done